### ISSUE TITLE
Offine for Moondeck Buddy

### ISSUE THREAD

**Issue by OuNiSang** [FrogTheFrog/moondeck-buddy#98]
Hi there

I followed all the settings on the wiki and managed to get one successful config when I first time implemented it, the game stream and Buddy both show online once 

However, after I rebooted the computer (Both Steamdeck and PC ), the Steamdeck side only has GameStream online, but the Buddy is kept Offline, and Piar with Buddy is grayed out (I did pair it in the first run) 

Then I started reinstalling the plugin on both steamdeck and PC, and I managed to get the GameStream online again. But the Buddy is still offline. Worth mentioning is that this time it does not require me to type in the Pair key again, even though I think we should clear out all the user settings when we uninstall it 

I did a check through PC on URL `https://localhost:59999/apiVersion` and it is working with turn 
``` JSON
{"version":4}
```

And here is the log that I have 

``` 
[19:58:45.551] INFO     buddy.utils: Log location: "C:/Users/****/AppData/Local/Programs/MoonDeckBuddy/bin/moondeckbuddy.log"
[19:58:45.552] INFO     buddy.main: startup. Version: 1.6.1
[19:58:45.696] INFO     buddy.server: Server started listening at port 59999
[19:58:45.696] INFO     buddy.main: startup finished.
[19:58:45.696] DEBUG    buddy.shared: getAppName() >>  "MoonDeckBuddy"
[19:58:45.696] DEBUG    buddy.shared: getLogDir() >>  "C:/Users/****/AppData/Local/Programs/MoonDeckBuddy/bin"
[19:58:45.697] DEBUG    buddy.shared: getLogName() >>  "moondeckbuddy.log"
[19:58:45.697] DEBUG    buddy.shared: getLogPath() >>  "C:/Users/13454/AppData/Local/Programs/MoonDeckBuddy/bin/moondeckbuddy.log"
[19:58:45.697] DEBUG    buddy.shared: getSettingsDir() >>  "C:/Users/****/AppData/Local/Programs/MoonDeckBuddy/bin"
[19:58:45.697] DEBUG    buddy.shared: getSettingsName() >>  "settings.json"
[19:58:45.698] DEBUG    buddy.shared: getSettingsPath() >>  "C:/Users/13454/AppData/Local/Programs/MoonDeckBuddy/bin/settings.json"
[19:58:45.698] DEBUG    buddy.shared: getAutoStartDir() >>  "C:/Users/****/AppData/Roaming/Microsoft/Windows/Start Menu/Programs/Startup"
[19:58:45.698] DEBUG    buddy.shared: getAutoStartPath() >>  "C:/Users/****/AppData/Roaming/Microsoft/Windows/Start Menu/Programs/Startup/MoonDeckBuddy.lnk"
[19:58:45.698] DEBUG    buddy.shared: getAutoStartExec() >>  "C:/Users/13454/AppData/Local/Programs/MoonDeckBuddy/bin/MoonDeckBuddy.exe"
[19:58:45.698] DEBUG    buddy.utils: Mapped "SecureProtocols" to 4
[19:58:45.699] INFO     buddy.os: Steam exec path: "c:/program files (x86)/steam/steam.exe"
[19:59:20.347] DEBUG    buddy.main: 
Request: QHttpServerRequest((Url: QUrl("https://localhost:59999/apiVersion"))(Headers: ("Host": "localhost:59999"), ("Connection": "keep-alive"), ("Cache-Control": "max-age=0"), ("sec-ch-ua": "\"Chromium\";v=\"128\", \"Not;A=Brand\";v=\"24\", \"Google Chrome\";v=\"128\""), ("sec-ch-ua-mobile": "?0"), ("sec-ch-ua-platform": "\"Windows\""), ("Upgrade-Insecure-Requests": "1"), ("User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"), ("Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"), ("Sec-Fetch-Site": "none"), ("Sec-Fetch-Mode": "navigate"), ("Sec-Fetch-User": "?1"), ("Sec-Fetch-Dest": "document"), ("Accept-Encoding": "gzip, deflate, br, zstd"), ("Accept-Language": "zh-CN,zh;q=0.9,en-US;q=0.8,en;q=0.7,zh-TW;q=0.6,zh-HK;q=0.5"))(RemoteHost: QHostAddress("::1"))(BodySize: 0)) | "" 
Response: QHttpServerResponder::StatusCode::Ok | "{\"version\":4}"

```

Not quite sure what should I do next, please let me know if you need more info on this.

**Comment by FrogTheFrog** [comment]
The IDs will be cleared out with with the new upcoming installer.

Can you check if the url is reachable from the SteamDeck? Also just try to start streaming and paste the /tmp/moondeck-runner log please.

**Comment by OuNiSang** [comment]
I just reinstalled the PC server side and the steamdeck side of the plugin, but looks like I dropped into the same issue again. 

looks like Steamdeck side would not erase the user data. Therefore I go to the desktop mod to delete the settings manually according to the instruction in ReadMe. However the status still shows offline after I clear the user data out 

Before I reset, this is the log file in from Steamdeck Client
[moondeck_Pre.log](https://github.com/user-attachments/files/17009028/moondeck.log)
Here is the Log file after I reset the user info 
[moondeck_Post.log](https://github.com/user-attachments/files/17009047/moondeck-2.log)

Solution: 

After that I double check the this [URL ](https://github.com/FrogTheFrog/moondeck-buddy/wiki/Troubleshooting#:~:text=https%3A//%3Chost_ip%3E%3A%3Cbuddy_port%3E/apiVersion), the API check URL in the wiki on steamdeck. Looks like it keep struggling responding, then I think that might be my firewall setting. 

After digging it a bit, it looks like I mark the moon deck only passed under private network. Then I delete all the rules and re-install the plugin on PC and let it pass the public network as well. Then I got it paired up. 

This looks like a problem solved for me
